akka-stream/src/main/scala/akka/stream/impl/Stages.scala

```diff
@@ -164,6 +165,11 @@ private[stream] object Stages {
     override def create(attr: Attributes): Stage[T, immutable.Seq[T]] = fusing.Grouped(n)
   }
 
+  final case class Limit[T](n: Int, attributes: Attributes = limit) extends SymbolicStage[T, T] {
+    require(n > 0, "n must be greater than 0")
```

Comment:
I think we can allow zero here. That will still allow completion without any elements.